Error while checking build/O2/o2-dataflow-cs8 for c45d930e80feb52e1f4c0555768ea87311759539 at 2022-09-05 11:27:
```
## sw/BUILD/O2-latest/log
100% tests passed, 0 tests failed out of 423
100% tests passed, 0 tests failed out of 101


## sw/BUILD/QualityControl-latest/log
23/42 Test #25: testCheckWorkflow .......................***Failed   10.91 sec
[2871:qc-task-TST-QYZTask]: 2022-09-05 09:26:05.775383 !!! [1;31mError - Could not find the DPL InfoLogger[0m
[2872:qc-task-TST-XYZTask]: 2022-09-05 09:26:05.868606 !!! [1;31mError - Could not find the DPL InfoLogger[0m
[2873:qc-task-TST-newName]: 2022-09-05 09:26:05.823215 !!! [1;31mError - Could not find the DPL InfoLogger[0m
[2874:qc-check-sink-QTST_QYZTask_0]: 2022-09-05 09:26:05.887875 !!! [1;31mError - Could not find the DPL InfoLogger.[0m
[2875:qc-check-sink-QTST_XYZTask_0]: 2022-09-05 09:26:05.868615 !!! [1;31mError - Could not find the DPL InfoLogger.[0m
[2876:qc-check-TST-u2F0]: 2022-09-05 09:26:05.891310 !!! [1;31mError - Could not find the DPL InfoLogger.[0m
[2877:qc-check-TST-CheckSeparately]: 2022-09-05 09:26:05.858048 !!! [1;31mError - Could not find the DPL InfoLogger.[0m
[2878:qc-check-TST-XYZCheck]: 2022-09-05 09:26:05.860524 !!! [1;31mError - Could not find the DPL InfoLogger.[0m
[ERROR] pid 2872 (qc-task-TST-XYZTask) crashed with 1
[ERROR] SEVERE: Device qc-task-TST-XYZTask (2872) returned with 1
98% tests passed, 1 tests failed out of 42
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9752/c45d930e80feb52e1f4c0555768ea87311759539/build_O2_o2-dataflow-cs8/pretty.html).

--- a/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
+++ b/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
@@ -172,6 +172,9 @@ class NoiseMap
   void merge(const NoiseMap* prev) {}
   const std::map<int, int>* getChipMap(int chip) const { return chip < (int)mNoisyPixels.size() ? &mNoisyPixels[chip] : nullptr; }
 
+  std::map<int, int>& getChip(int chip) { return mNoisyPixels[chip]; }
+  const std::map<int, int>& getChip(int chip) const { return mNoisyPixels[chip]; }
+
   void maskFullChip(int chip, bool cleanNoisyPixels = false)
   {
     if (cleanNoisyPixels) {
@@ -199,6 +202,7 @@ class NoiseMap
     return std::ceil((1. + 1. / t) / (relErr * relErr));
   }
 
+  size_t size() const { return mNoisyPixels.size(); }
   void setNumOfStrobes(long n) { mNumOfStrobes = n; }
   void addStrobes(long n) { mNumOfStrobes += n; }
   long getNumberOfStrobes() const { return mNumOfStrobes; }

--- a/Detectors/ITSMFT/ITS/calibration/include/ITSCalibration/NoiseCalibrator.h
+++ b/Detectors/ITSMFT/ITS/calibration/include/ITSCalibration/NoiseCalibrator.h
@@ -18,6 +18,7 @@
 #include "DataFormatsITSMFT/TopologyDictionary.h"
 #include "DataFormatsITSMFT/NoiseMap.h"
 #include "ITSMFTReconstruction/PixelData.h"
+#include "ITSMFTReconstruction/ChipMappingITS.h"
 #include "gsl/span"
 
 namespace o2
@@ -35,6 +36,8 @@ namespace its
 class NoiseCalibrator
 {
  public:
+  static constexpr int NChips = o2::itsmft::ChipMappingITS::getNChips();
+
   NoiseCalibrator() = default;
   NoiseCalibrator(bool one, float prob, float relErr = 0.2) : m1pix(one), mProbabilityThreshold(prob), mProbRelErr(relErr)
   {
@@ -50,34 +53,38 @@ class NoiseCalibrator
   bool processTimeFrameDigits(gsl::span<const o2::itsmft::Digit> const& digits,
                               gsl::span<const o2::itsmft::ROFRecord> const& rofs);
 
+  void addMap(const o2::itsmft::NoiseMap& extMap);
+
   void finalize(float cutIB = -1.);
 
   void setNThreads(int n) { mNThreads = n > 0 ? n : 1; }
 
   void setMinROFs(long n) { mMinROFs = n; }
+  long getMinROFs() const { return mMinROFs; }
 
   void setClusterDictionary(const o2::itsmft::TopologyDictionary* d) { mDict = d; }
 
   const o2::itsmft::NoiseMap& getNoiseMap() const { return mNoiseMap; }
 
-  void setInstanceID(size_t i) { mInstanceID = i; }
-  void setNInstances(size_t n) { mNInstances = n; }
+  void setInstanceID(int i) { mInstanceID = i; }
+  void setNInstances(int n) { mNInstances = n; }
   auto getInstanceID() const { return mInstanceID; }
   auto getNInstances() const { return mNInstances; }
+  auto getNStrobes() const { return mNumberOfStrobes; }
 
  private:
   const o2::itsmft::TopologyDictionary* mDict = nullptr;
-  o2::itsmft::NoiseMap mNoiseMap{24120};
+  o2::itsmft::NoiseMap mNoiseMap{NChips};
   float mProbabilityThreshold = 3e-6f;
   float mProbRelErr = 0.2; // relative error on channel noise to apply the threshold
   long mMinROFs = 0;
   unsigned int mNumberOfStrobes = 0;
   bool m1pix = true;
   int mNThreads = 1;
-  size_t mInstanceID = 0; // pipeline instance
-  size_t mNInstances = 1; // total number of pipelines
+  int mInstanceID = 0; // pipeline instance
+  int mNInstances = 1; // total number of pipelines
   std::vector<int> mChipIDs;
-  std::array<std::vector<int>, 24120> mChipHits;
+  std::array<std::vector<int>, NChips> mChipHits;
 };
 
 } // namespace its

--- a/Detectors/ITSMFT/ITS/calibration/include/ITSCalibration/NoiseCalibratorSpec.h
+++ b/Detectors/ITSMFT/ITS/calibration/include/ITSCalibration/NoiseCalibratorSpec.h
@@ -41,7 +41,11 @@ namespace its
 class NoiseCalibratorSpec : public Task
 {
  public:
-  NoiseCalibratorSpec(bool useClusters = false, std::shared_ptr<o2::base::GRPGeomRequest> req = {}) : mCCDBRequest(req), mUseClusters(useClusters)
+  enum ProcessingMode { Full,
+                        Accumulate,
+                        Normalize };
+
+  NoiseCalibratorSpec(ProcessingMode md = ProcessingMode::Full, bool useClusters = false, std::shared_ptr<o2::base::GRPGeomRequest> req = {}) : mMode(md), mCCDBRequest(req), mUseClusters(useClusters)
   {
     mTimer.Stop();
   }
@@ -51,27 +55,32 @@ class NoiseCalibratorSpec : public Task
   void run(ProcessingContext& pc) final;
   void endOfStream(EndOfStreamContext& ec) final;
   void finaliseCCDB(ConcreteDataMatcher& matcher, void* obj) final;
+  void setProcessingMode(ProcessingMode m) { mMode = m; }
+  ProcessingMode getProcessingMode() const { return mMode; }
 
  private:
   void addDatabaseEntry(int chip, int row, int col);
   void sendOutput(DataAllocator& output);
+  void sendAccumulatedMap(DataAllocator& output);
   void updateTimeDependentParams(ProcessingContext& pc);
   std::unique_ptr<CALIBRATOR> mCalibrator = nullptr;
   std::shared_ptr<o2::base::GRPGeomRequest> mCCDBRequest;
   size_t mDataSizeStat = 0;
   size_t mNClustersProc = 0;
   int mValidityDays = 3;
+  int mNPartsDone = 0; // number of accumalated parts in Normalization mode
   bool mUseClusters = false;
   bool mStopMeOnly = false; // send QuitRequest::Me instead of QuitRequest::All
   TStopwatch mTimer{};
   float mNoiseCutIB = -1.;
   o2::dcs::DCSconfigObject_t mNoiseMapDCS; // noisy pixels to be sent to DCS CCDB
   std::vector<int>* mConfDBmap{nullptr};
+  ProcessingMode mMode = ProcessingMode::Full;
 };
 
 /// create a processor spec
 /// run ITS noise calibration
-DataProcessorSpec getNoiseCalibratorSpec(bool useClusters);
+DataProcessorSpec getNoiseCalibratorSpec(bool useClusters, int pmode = 0);
 
 } // namespace its
 } // namespace o2

--- a/Detectors/ITSMFT/ITS/calibration/src/NoiseCalibrator.cxx
+++ b/Detectors/ITSMFT/ITS/calibration/src/NoiseCalibrator.cxx
@@ -133,6 +133,21 @@ bool NoiseCalibrator::processTimeFrameDigits(gsl::span<const o2::itsmft::Digit> 
   return (mNumberOfStrobes > mMinROFs) ? true : false;
 }
 
+void NoiseCalibrator::addMap(const o2::itsmft::NoiseMap& extMap)
+{
+  // add preproprecessed map to total
+#ifdef WITH_OPENMP
+#pragma omp parallel for schedule(dynamic) num_threads(mNThreads)
+#endif
+  for (int ic = 0; ic < NChips; ic++) {
+    const auto& chExt = extMap.getChip(ic);
+    auto& chCurr = mNoiseMap.getChip(ic);
+    for (auto it : chExt) {
+      chCurr[it.first] += it.second;
+    }
+  }
+}
+
 void NoiseCalibrator::finalize(float cutIB)
 {
   LOG(info) << "Number of processed strobes is " << mNumberOfStrobes;

--- a/Detectors/ITSMFT/ITS/calibration/src/NoiseCalibratorSpec.cxx
+++ b/Detectors/ITSMFT/ITS/calibration/src/NoiseCalibratorSpec.cxx
@@ -18,6 +18,7 @@
 #include "DataFormatsITSMFT/CompCluster.h"
 #include "DataFormatsITSMFT/ROFRecord.h"
 #include "DataFormatsITSMFT/TopologyDictionary.h"
+#include "ITSMFTReconstruction/ChipMappingITS.h"
 
 #include "Framework/Logger.h"
 #include "Framework/ControlService.h"
@@ -45,7 +46,6 @@ void NoiseCalibratorSpec::init(InitContext& ic)
   mStopMeOnly = ic.options().get<bool>("stop-me-only");
   mCalibrator = std::make_unique<CALIBRATOR>(onepix, probT, probTRelErr);
   mCalibrator->setNThreads(ic.options().get<int>("nthreads"));
-
   mValidityDays = ic.options().get<int>("validity-days");
   if (mValidityDays < 1) {
     mValidityDays = 1;
@@ -57,36 +57,70 @@ void NoiseCalibratorSpec::run(ProcessingContext& pc)
   updateTimeDependentParams(pc);
   mTimer.Start(false);
   static bool firstCall = true;
+  static bool done = false;
+  if (done) {
+    return;
+  }
   if (firstCall) {
     firstCall = false;
-    mCalibrator->setInstanceID(pc.services().get<const o2::framework::DeviceSpec>().inputTimesliceId);
-    mCalibrator->setNInstances(pc.services().get<const o2::framework::DeviceSpec>().maxInputTimeslices);
+    mCalibrator->setInstanceID((int)pc.services().get<const o2::framework::DeviceSpec>().inputTimesliceId);
+    mCalibrator->setNInstances((int)pc.services().get<const o2::framework::DeviceSpec>().maxInputTimeslices);
+    if (mMode == ProcessingMode::Accumulate) {
+      mCalibrator->setMinROFs(mCalibrator->getMinROFs() / mCalibrator->getNInstances());
+    }
   }
-  bool done = false;
-  if (mUseClusters) {
-    const auto compClusters = pc.inputs().get<gsl::span<o2::itsmft::CompClusterExt>>("compClusters");
-    gsl::span<const unsigned char> patterns = pc.inputs().get<gsl::span<unsigned char>>("patterns");
-    const auto rofs = pc.inputs().get<gsl::span<o2::itsmft::ROFRecord>>("ROframes");
-    mNClustersProc += compClusters.size();
-    mDataSizeStat +=
-      rofs.size() * sizeof(o2::itsmft::ROFRecord) + patterns.size() +
-      compClusters.size() * sizeof(o2::itsmft::CompClusterExt);
-    done = mCalibrator->processTimeFrameClusters(compClusters, patterns, rofs);
+
+  if (mMode == ProcessingMode::Full || mMode == ProcessingMode::Accumulate) {
+    if (mUseClusters) {
+      const auto compClusters = pc.inputs().get<gsl::span<o2::itsmft::CompClusterExt>>("compClusters");
+      gsl::span<const unsigned char> patterns = pc.inputs().get<gsl::span<unsigned char>>("patterns");
+      const auto rofs = pc.inputs().get<gsl::span<o2::itsmft::ROFRecord>>("ROframes");
+      mNClustersProc += compClusters.size();
+      mDataSizeStat +=
+        rofs.size() * sizeof(o2::itsmft::ROFRecord) + patterns.size() +
+        compClusters.size() * sizeof(o2::itsmft::CompClusterExt);
+      done = mCalibrator->processTimeFrameClusters(compClusters, patterns, rofs);
+    } else {
+      const auto digits = pc.inputs().get<gsl::span<o2::itsmft::Digit>>("digits");
+      const auto rofs = pc.inputs().get<gsl::span<o2::itsmft::ROFRecord>>("ROframes");
+      mNClustersProc += digits.size();
+      mDataSizeStat += digits.size() * sizeof(o2::itsmft::Digit) + rofs.size() * sizeof(o2::itsmft::ROFRecord);
+      done = mCalibrator->processTimeFrameDigits(digits, rofs);
+    }
   } else {
-    const auto digits = pc.inputs().get<gsl::span<o2::itsmft::Digit>>("digits");
-    const auto rofs = pc.inputs().get<gsl::span<o2::itsmft::ROFRecord>>("ROframes");
-    mNClustersProc += digits.size();
-    mDataSizeStat += digits.size() * sizeof(o2::itsmft::Digit) + rofs.size() * sizeof(o2::itsmft::ROFRecord);
-    done = mCalibrator->processTimeFrameDigits(digits, rofs);
+    const auto extMap = pc.inputs().get<o2::itsmft::NoiseMap*>("mapspart");
+    gsl::span<const int> partInfo = pc.inputs().get<gsl::span<int>>("mapspartInfo");
+    mCalibrator->addMap(*extMap.get());
+    done = (++mNPartsDone == partInfo[1]);
+    LOGP(info, "Received accumulated map with {} of {}, total number of maps = {}", partInfo[0], partInfo[1], mNPartsDone);
   }
   if (done) {
-    pc.inputs().get<std::vector<int>*>("confdbmap"); // to trigger finaliseCCDB
     LOG(info) << "Minimum number of noise counts has been reached !";
-    sendOutput(pc.outputs());
-    pc.services().get<ControlService>().readyToQuit(mStopMeOnly ? QuitRequest::Me : QuitRequest::All);
+    if (mMode == ProcessingMode::Full || mMode == ProcessingMode::Normalize) {
+      sendOutput(pc.outputs());
+      pc.services().get<ControlService>().readyToQuit(mStopMeOnly ? QuitRequest::Me : QuitRequest::All);
+    } else {
+      sendAccumulatedMap(pc.outputs());
+      pc.services().get<o2::framework::ControlService>().endOfStream();
+    }
   }
 
   mTimer.Stop();
+}
+
+void NoiseCalibratorSpec::sendAccumulatedMap(DataAllocator& output)
+{
+  static bool done = false;
+  if (done) {
+    return;
+  }
+  done = true;
+  output.snapshot(Output{"ITS", "NOISEMAPPART", (unsigned int)mCalibrator->getInstanceID()}, mCalibrator->getNoiseMap());
+  std::vector<int> outInf;
+  outInf.push_back(mCalibrator->getInstanceID());
+  outInf.push_back(mCalibrator->getNInstances());
+  output.snapshot(Output{"ITS", "NOISEMAPPARTINF", (unsigned int)mCalibrator->getInstanceID()}, outInf);
+  LOGP(info, "Sending accumulated map with {} ROFs processed", mCalibrator->getNStrobes());
 }
 
 void NoiseCalibratorSpec::sendOutput(DataAllocator& output)
@@ -122,7 +156,7 @@ void NoiseCalibratorSpec::sendOutput(DataAllocator& output)
   LOG(info) << "sending of o2::itsmft::NoiseMap done";
 
   // Preparing the object for DCS CCDB and sending it to output
-  for (int ichip = 0; ichip < 24120; ichip++) {
+  for (int ichip = 0; ichip < o2::itsmft::ChipMappingITS::getNChips(); ichip++) {
     const std::map<int, int>* chipmap = payload.getChipMap(ichip);
     if (chipmap) {
       for (std::map<int, int>::const_iterator it = chipmap->begin(); it != chipmap->end(); ++it) {
@@ -156,7 +190,11 @@ void NoiseCalibratorSpec::addDatabaseEntry(int chip, int row, int col)
 
 void NoiseCalibratorSpec::endOfStream(o2::framework::EndOfStreamContext& ec)
 {
-  sendOutput(ec.outputs());
+  if (mMode == ProcessingMode::Accumulate) {
+    sendAccumulatedMap(ec.outputs());
+  } else {
+    sendOutput(ec.outputs());
+  }
 }
 
 ///_______________________________________
@@ -169,7 +207,9 @@ void NoiseCalibratorSpec::updateTimeDependentParams(ProcessingContext& pc)
     if (mUseClusters) {
       pc.inputs().get<o2::itsmft::TopologyDictionary*>("cldict"); // just to trigger the finaliseCCDB
     }
-    pc.inputs().get<std::vector<int>*>("confdbmap");
+    if (mMode == ProcessingMode::Full || mMode == ProcessingMode::Normalize) {
+      pc.inputs().get<std::vector<int>*>("confdbmap");
+    }
   }
 }
 
@@ -190,20 +230,48 @@ void NoiseCalibratorSpec::finaliseCCDB(ConcreteDataMatcher& matcher, void* obj)
   }
 }
 
-DataProcessorSpec getNoiseCalibratorSpec(bool useClusters)
+DataProcessorSpec getNoiseCalibratorSpec(bool useClusters, int pmode)
 {
-  std::vector<InputSpec> inputs;
-  if (useClusters) {
-    inputs.emplace_back("compClusters", "ITS", "COMPCLUSTERS", 0, Lifetime::Timeframe);
-    inputs.emplace_back("patterns", "ITS", "PATTERNS", 0, Lifetime::Timeframe);
-    inputs.emplace_back("ROframes", "ITS", "CLUSTERSROF", 0, Lifetime::Timeframe);
-    inputs.emplace_back("cldict", "ITS", "CLUSDICT", 0, Lifetime::Condition, ccdbParamSpec("ITS/Calib/ClusterDictionary"));
+  NoiseCalibratorSpec::ProcessingMode md = NoiseCalibratorSpec::ProcessingMode::Full;
+  std::string name = "its-noise-calibrator";
+  if (pmode == int(NoiseCalibratorSpec::ProcessingMode::Full)) {
+    md = NoiseCalibratorSpec::ProcessingMode::Full;
+  } else if (pmode == int(NoiseCalibratorSpec::ProcessingMode::Accumulate)) {
+    md = NoiseCalibratorSpec::ProcessingMode::Accumulate;
+  } else if (pmode == int(NoiseCalibratorSpec::ProcessingMode::Normalize)) {
+    md = NoiseCalibratorSpec::ProcessingMode::Normalize;
+    name = "its-noise-calibrator_Norm";
   } else {
-    inputs.emplace_back("digits", "ITS", "DIGITS", 0, Lifetime::Timeframe);
-    inputs.emplace_back("ROframes", "ITS", "DIGITSROF", 0, Lifetime::Timeframe);
+    LOG(fatal) << "Unknown processing mode " << pmode;
   }
-  inputs.emplace_back("confdbmap", "ITS", "CONFDBMAP", 0, Lifetime::Condition, ccdbParamSpec("ITS/Calib/Confdbmap"));
 
+  std::vector<InputSpec> inputs;
+  std::vector<OutputSpec> outputs;
+
+  if (md == NoiseCalibratorSpec::ProcessingMode::Full || md == NoiseCalibratorSpec::ProcessingMode::Accumulate) {
+    if (useClusters) {
+      inputs.emplace_back("compClusters", "ITS", "COMPCLUSTERS", 0, Lifetime::Timeframe);
+      inputs.emplace_back("patterns", "ITS", "PATTERNS", 0, Lifetime::Timeframe);
+      inputs.emplace_back("ROframes", "ITS", "CLUSTERSROF", 0, Lifetime::Timeframe);
+      inputs.emplace_back("cldict", "ITS", "CLUSDICT", 0, Lifetime::Condition, ccdbParamSpec("ITS/Calib/ClusterDictionary"));
+    } else {
+      inputs.emplace_back("digits", "ITS", "DIGITS", 0, Lifetime::Timeframe);
+      inputs.emplace_back("ROframes", "ITS", "DIGITSROF", 0, Lifetime::Timeframe);
+    }
+  } else {
+    useClusters = false;                                                                                         // not needed for normalization
+    inputs.emplace_back("mapspart", ConcreteDataTypeMatcher{"ITS", "NOISEMAPPART"}, Lifetime::Timeframe);        // for normalization of multiple inputs only
+    inputs.emplace_back("mapspartInfo", ConcreteDataTypeMatcher{"ITS", "NOISEMAPPARTINF"}, Lifetime::Timeframe); // for normalization of multiple inputs only
+  }
+  if (md == NoiseCalibratorSpec::ProcessingMode::Full || md == NoiseCalibratorSpec::ProcessingMode::Normalize) {
+    inputs.emplace_back("confdbmap", "ITS", "CONFDBMAP", 0, Lifetime::Condition, ccdbParamSpec("ITS/Calib/Confdbmap"));
+
+    outputs.emplace_back(ConcreteDataTypeMatcher{o2::calibration::Utils::gDataOriginCDBPayload, "ITS_NOISE"}, Lifetime::Sporadic);
+    outputs.emplace_back(ConcreteDataTypeMatcher{o2::calibration::Utils::gDataOriginCDBWrapper, "ITS_NOISE"}, Lifetime::Sporadic);
+  } else { // in accumulation mode the output is a map
+    outputs.emplace_back(ConcreteDataTypeMatcher{"ITS", "NOISEMAPPART"}, Lifetime::Sporadic);
+    outputs.emplace_back(ConcreteDataTypeMatcher{"ITS", "NOISEMAPPARTINF"}, Lifetime::Sporadic);
+  }
   auto ccdbRequest = std::make_shared<o2::base::GRPGeomRequest>(false,                          // orbitResetTime
                                                                 false,                          // GRPECS=true
                                                                 false,                          // GRPLHCIF
@@ -211,15 +279,12 @@ DataProcessorSpec getNoiseCalibratorSpec(bool useClusters)
                                                                 false,                          // askMatLUT
                                                                 o2::base::GRPGeomRequest::None, // geometry
                                                                 inputs);
-  std::vector<OutputSpec> outputs;
-  outputs.emplace_back(ConcreteDataTypeMatcher{o2::calibration::Utils::gDataOriginCDBPayload, "ITS_NOISE"}, Lifetime::Sporadic);
-  outputs.emplace_back(ConcreteDataTypeMatcher{o2::calibration::Utils::gDataOriginCDBWrapper, "ITS_NOISE"}, Lifetime::Sporadic);
 
   return DataProcessorSpec{
-    "its-noise-calibrator",
+    name,
     inputs,
     outputs,
-    AlgorithmSpec{adaptFromTask<NoiseCalibratorSpec>(useClusters, ccdbRequest)},
+    AlgorithmSpec{adaptFromTask<NoiseCalibratorSpec>(md, useClusters, ccdbRequest)},
     Options{
       {"1pix-only", VariantType::Bool, false, {"Fast 1-pixel calibration only (cluster input only)"}},
       {"prob-threshold", VariantType::Float, 3.e-6f, {"Probability threshold for noisy pixels"}},

--- a/Detectors/ITSMFT/ITS/calibration/testWorkflow/its-noise-calib-workflow.cxx
+++ b/Detectors/ITSMFT/ITS/calibration/testWorkflow/its-noise-calib-workflow.cxx
@@ -19,6 +19,7 @@ void customize(std::vector<ConfigParamSpec>& workflowOptions)
 {
   // option allowing to set parameters
   workflowOptions.push_back(ConfigParamSpec{"use-clusters", VariantType::Bool, false, {"Use clusters instead of digits"}});
+  workflowOptions.push_back(ConfigParamSpec{"processing-mode", VariantType::Int, 0, {"processing mode: 0 - accumulate & normalize, 1 - accumulated and send w/o normalization, 2 - receive maps from mode 1 accumulators and normalized"}});
   workflowOptions.push_back(ConfigParamSpec{"configKeyValues", VariantType::String, "", {"Semicolon separated key=value strings"}});
 }
 
@@ -32,6 +33,6 @@ WorkflowSpec defineDataProcessing(ConfigContext const& configcontext)
 {
   WorkflowSpec specs;
   o2::conf::ConfigurableParam::updateFromString(configcontext.options().get<std::string>("configKeyValues"));
-  specs.emplace_back(o2::its::getNoiseCalibratorSpec(configcontext.options().get<bool>("use-clusters")));
+  specs.emplace_back(o2::its::getNoiseCalibratorSpec(configcontext.options().get<bool>("use-clusters"), configcontext.options().get<int>("processing-mode")));
   return specs;
 }

